src/MainWindow.cpp

```diff
@@ -4362,71 +4308,83 @@ void MainWindow::on_toolTemplatesLoadAction_triggered()
-        QCoreApplication::removeTranslator(qtTranslator);
-    }
-    if (merkaartorTranslator) {
-        QCoreApplication::removeTranslator(merkaartorTranslator);
-    }
-    QString DefaultLanguage = getDefaultLanguage();
-    if (DefaultLanguage != "-" && DefaultLanguage != "en")
-    {
+QString MainWindow::translationsPath() {
+    return QCoreApplication::applicationDirPath() + "/translations";
+}
+
 #if defined(Q_OS_MAC)
-        QDir resources = QDir(QCoreApplication::applicationDirPath());
-        resources.cdUp();
-        resources.cd("Resources");
+QString MainWindow::macOsTranslationsPath() {
+    QDir resources = QDir(QCoreApplication::applicationDirPath());
+    resources.cdUp();
+    resources.cd("Resources");
+    return resources.absolutePath();
+}
 #endif
-        qtTranslator = new QTranslator;
-        bool retQt;
-        if (g_Merk_Portable) {
-            retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), QCoreApplication::applicationDirPath() + "/translations");
-        } else {
+
+bool MainWindow::tryLoadQtTranslator(const QString& languagePrefix)
+{
+    if (g_Merk_Portable) {
+        return qtTranslator->load("qt_" + languagePrefix, translationsPath());
+    } else {
 #ifdef TRANSDIR_SYSTEM
-            if (!QDir::isAbsolutePath(STRINGIFY(TRANSDIR_SYSTEM)))
-                retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), QCoreApplication::applicationDirPath() + "/" + STRINGIFY(TRANSDIR_SYSTEM));
-            else
-                retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), STRINGIFY(TRANSDIR_SYSTEM));
+        if (!QDir::isAbsolutePath(STRINGIFY(TRANSDIR_SYSTEM)))
+            return qtTranslator->load("qt_" + languagePrefix, QCoreApplication::applicationDirPath() + "/" + STRINGIFY(TRANSDIR_SYSTEM));
+        else
+            return qtTranslator->load("qt_" + languagePrefix, STRINGIFY(TRANSDIR_SYSTEM));
 #else
 #if defined(Q_OS_MAC)
-            retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), resources.absolutePath());
+        return qtTranslator->load("qt_" + languagePrefix, macOsTranslationsPath());
 #else
-            retQt = qtTranslator->load("qt_" + DefaultLanguage.left(2), QLibraryInfo::location(QLibraryInfo::TranslationsPath));
+        return qtTranslator->load("qt_" + languagePrefix, QLibraryInfo::location(QLibraryInfo::TranslationsPath));
 #endif // Q_OS_MAC
 #endif // TRANSDIR_SYSTEM
-        }
-
-        if (retQt)
-            QCoreApplication::installTranslator(qtTranslator);
-
-        merkaartorTranslator = new QTranslator;
-        // Do not prevent Merkaartor translations to be loaded, even if there is no Qt translation for the language.
+    }
+}
 
-        bool retM;
-        if (g_Merk_Portable) {
-            retM = merkaartorTranslator->load("merkaartor_" + DefaultLanguage, QCoreApplication::applicationDirPath() + "/translations");
-        } else {
+bool MainWindow::tryLoadMerkaartorTranslator(const QString& language)
```

Comment:
Same as `tryLoadQtTranslator` - please check the control flow here, especially with the different values of `g_Merk_Portable`, `Q_OS_MAC` and `TRANSDIR_MERKAARTOR`.